web/META-INF/resources/js/controllers.js

```diff
@@ -0,0 +1,234 @@
+            // initialize to page 1
+            vm.setPage(1);
+        }
+
+        function setPage(page) {
+            if (page < 1 || page > vm.pager.totalPages) {
+                return;
+            }
+
+            // get pager object from service
+            vm.pager = PagerService.GetPager(vm.PDItems.length, page);
+
+            // get current page of items
+            vm.items = vm.PDItems.slice(vm.pager.startIndex, vm.pager.endIndex + 1);
+        }
+
+        this.searchTopic = function(topic) {
+            var searchKeyWords = new String();
+
+            if (topic.search(',') != -1 ) {
+                var topics = topic.split(',');
+                for (var i = 0; i < topics.length; i++) {
+                    searchKeyWords += topics[i];
+                }
+            } else {
+                searchKeyWords = topic;
+            } 
+            
+            if (searchKeyWords.search('/') != -1) {
+                var topics = searchKeyWords.split('/');
+                searchKeyWords = "";
+                for (var i = 0; i < topics.length; i++) {
+                    searchKeyWords += topics[i];
+                }
+            } 
+            
+            $rootScope.searchOptions.term = searchKeyWords;
+            MetaData.get({query: searchKeyWords, operator: opt}, 
+                function success(response) {
+                    vm.PDItems = response.PDResults;
+                    vm.totalMatches = vm.PDItems.length;
+                    initController();
+                    //$scope.PDResults = response.PDResults;
+
+                },
+                function error(errorResponse) {
+                    console.log("Error:" + JSON.stringify(errorResponse));
+                }
+            );
+        }
+
+        MetaData.get({query: word, operator: opt}, 
+                function success(response) {
+                    vm.PDItems = response.PDResults;
+                    vm.totalMatches = vm.PDItems.length;
+                    initController();
+                    //$scope.PDResults = response.PDResults;
+
+                },
+                function error(errorResponse) {
+                    console.log("Error:" + JSON.stringify(errorResponse));
+                }
+        );
+    }]);
+
+mudrodControllers.controller('datasetViewCtrl', ['$scope', '$routeParams', 'DatasetDetail',
+    function datasetViewCtrl($scope, $routeParams, DatasetDetail) {
+        var shortname = $routeParams.shortname;   
+
+        DatasetDetail.get({shortname: shortname}, 
+                function success(response) {
+                    var dataAccessUrls;
+                    $scope.dataset = response.PDResults[0];
+                    var dataAccessUrl = $scope.dataset['DatasetLocationPolicy-BasePath'];
+                    if(dataAccessUrl.search(',') != -1) {
+                        dataAccessUrls =  dataAccessUrl.split(',');
+                    }
+                    $scope.ftpUrl = dataAccessUrls[0];
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') TypeError can be thrown as "dataAccessUrls" might be null or undefined here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=javascript%3AS2259)